homeassistant/components/mqtt/mixins.py

```
@@ -0,0 +1,471 @@
+        await super().async_added_to_hass()
+        await self._availability_subscribe_topics()
+        self.async_on_remove(
+            async_dispatcher_connect(self.hass, MQTT_CONNECTED, self.async_mqtt_connect)
+        )
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, MQTT_DISCONNECTED, self.async_mqtt_connect
+            )
+        )
+
+    async def availability_discovery_update(self, config: dict):
+        """Handle updated discovery message."""
+        self._availability_setup_from_config(config)
+        await self._availability_subscribe_topics()
+
+    def _availability_setup_from_config(self, config):
+        """(Re)Setup."""
+        self._avail_topics = {}
+        if CONF_AVAILABILITY_TOPIC in config:
+            self._avail_topics[config[CONF_AVAILABILITY_TOPIC]] = {
+                CONF_PAYLOAD_AVAILABLE: config[CONF_PAYLOAD_AVAILABLE],
+                CONF_PAYLOAD_NOT_AVAILABLE: config[CONF_PAYLOAD_NOT_AVAILABLE],
+            }
+
+        if CONF_AVAILABILITY in config:
+            for avail in config[CONF_AVAILABILITY]:
+                self._avail_topics[avail[CONF_TOPIC]] = {
+                    CONF_PAYLOAD_AVAILABLE: avail[CONF_PAYLOAD_AVAILABLE],
+                    CONF_PAYLOAD_NOT_AVAILABLE: avail[CONF_PAYLOAD_NOT_AVAILABLE],
+                }
+
+        self._avail_config = config
+
+    async def _availability_subscribe_topics(self):
+        """(Re)Subscribe to topics."""
+
+        @callback
+        @log_messages(self.hass, self.entity_id)
+        def availability_message_received(msg: Message) -> None:
+            """Handle a new received MQTT availability message."""
+            topic = msg.topic
+            if msg.payload == self._avail_topics[topic][CONF_PAYLOAD_AVAILABLE]:
+                self._available = True
+            elif msg.payload == self._avail_topics[topic][CONF_PAYLOAD_NOT_AVAILABLE]:
+                self._available = False
+
```

What happens if the payload is none of these two ?

---

The MQTT components are a bit inconsistent, some components silently ignore unexpected data, some warn.
In this case nothing happens. Changing the behavior would be a breaking change.